SignalServiceKit/src/Storage/AxolotlStore/OWSPrimaryStorage+SessionStore.m

```diff
@@ -88,6 +88,9 @@ - (void)storeSession:(NSString *)contactIdentifier
     OWSAssertDebug(contactIdentifier.length > 0);
     OWSAssertDebug(deviceId >= 0);
     OWSAssertDebug([protocolContext isKindOfClass:[YapDatabaseReadWriteTransaction class]]);
+    if (!CurrentAppContext().isMainApp) {
```

Comment:
This could use a comment explaining why this is needed

Reply:
Oh yes, to decrypt a message twice in Notification Service Extension and the main app respectively, we skip the saving for Session Record if it is in the Notification Service Extension. So the message won't be marked as duplicated in the main app when we try to decrypt it.